src/main/java/org/jabref/gui/util/Theme.java

```diff
@@ -0,0 +1,124 @@
+package org.jabref.gui.util;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+
+import javafx.scene.Scene;
+
+import org.jabref.gui.JabRefFrame;
+import org.jabref.model.util.FileUpdateMonitor;
+import org.jabref.preferences.AppearancePreferences;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Installs the style file and provides live reloading.
+ * JabRef provides two inbuilt themes and a user customizable one: Light, Dark and Custom. The Light theme is basically
+ * the base.css theme. Every other theme is loaded as an addition to base.css.
+ */
+public enum Theme {
+    LIGHT("Light", ""),
+    DARK("Dark", "Dark.css"),
+    CUSTOM("Custom", "");
+
+    public static final String BASE_CSS = "Base.css";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Theme.class);
+
+    private static Optional<URL> additionalCssToLoad = Optional.empty(); // default: Light theme
+    private static FileUpdateMonitor fileUpdateMonitor;
+
+    private String name;
+    private Path additionalCssPath;
+
+    Theme(String name, String path) {
+        this.name = name;
+        this.additionalCssPath = Path.of(path);
+    }
+
+    public static void initialize(FileUpdateMonitor fileUpdateMonitor, PreferencesService preferences) {
```

Comment:
To be honest, I'm not a big fan of these static methods. Static methods should only be used when there is no internal state to manage. So for example its ok that the methods in StringUtil are static, as they mostly follow the pattern that they take an input, process it and return something.

However, here, there is a state that is managed: the fileUpdateMonitor and the additionalCssToLoad private fields. This leads to the hidden convention, that it's only allowed to call the installCss method after initalize is called. It's better to make this requirement explicit by making initalize a constructor of a normal class. This also has the advantage that you can mock/replace the themeloader in tests or for whatever reason. (I agree that it's somewhat academic in this particular case for the themeloader, as its not widely used. But it's good to follow the practice that dependencies are explicitly stated and not hidden in untestable static classes).

Reply:
Ok, I have reworked the class, the static objects are now part of the theme object. Should feel now somewhat like a cell factory with 'fluent' api (`getTheme().install.Css(scene)`). The fileUpdateMonitor dependency is still bound to the Globals object, but I did not have the time nor the energy to refactor everything now, so I would like to stop here and to leave the refactor of the Globals object for sometime later, as I really would like to finish the preferences refactor project first.